### PR TITLE
Include flag --with-build-config=bootstrap-debug

### PR DESCRIPTION
**Pull request by leonardoobaptistaa**
Build with Xcode 6.3 is broken. This PR is just a update from the original repo.
Original PR: https://github.com/Homebrew/homebrew/pull/38596
Explanation: https://github.com/mriddle/puppet-gcc/releases/tag/2.2.1

### CODE REVIEW

**Comment by DomT4** [comment]
@mikemcquaid Any idea what this error is about on Mountain Lion's bot?

```
==> Bottling gcc48-4.8.4.mountain_lion.bottle.1.tar.gz...
error: /usr/bin/install_name_tool: changing install names or rpaths can't be redone for: /usr/local/Cellar/gcc48/4.8.4/libexec/gcc/x86_64-apple-darwin12.6.0/4.8.4/lto1 (for architecture x86_64) because larger updated load commands do not fit (the program must be relinked, and you may need to use -headerpad or -headerpad_max_install_names)
Error: Failure while executing: /usr/bin/install_name_tool -change /usr/local/opt/gmp4/lib/libgmp.3.dylib @@HOMEBREW_PREFIX@@/opt/gmp4/lib/libgmp.3.dylib /usr/local/Cellar/gcc48/4.8.4/libexec/gcc/x86_64-apple-darwin12.6.0/4.8.4/lto1
```

Never seen anything like it, and it's only impacted ML's bot.

**Comment by mikemcquaid** [comment]
Yeh, weird. Looks like it'll need built with a different `LDFLAG` as suggested.

**Comment by DomT4** [comment]
A different ` LDFLAG ` pointing to where? Apologies if that's a *super* dumb question.

**Comment by mikemcquaid** [comment]
Adding `-headerpad` or `-headerpad_max_install_names`. You'll need to check the docs/grep other formulae for usage.

**Comment by DomT4** [comment]
It *looks* like it's ` ENV.append "LDFLAGS", "-headerpad_max_install_names" ` - From usage in Homebrew/homebrew, at least.

**Comment by mikemcquaid** [comment]
Sounds about right.

**Comment by DomT4** [comment]
@leonardoobaptistaa Can you try adding my comment above to ` def install ` and see if that placates the Mountain Lion build? Thanks!

**Comment by leonardoobaptistaa** [comment]
OK, done!
But I can't check if worked, running Yosemite here.

**Comment by DomT4** [comment]
No worries, the test-bot will punch through it on the 3 most recent OS X major versions at some point today (Bit of a backlog right now). Thanks for the change!

File: gcc48.rb [hunk 1]
```diff
@@ -57,6 +57,8 @@ def osmajor
 
   # GCC bootstraps itself, so it is OK to have an incompatible C++ stdlib
   cxxstdlib_check :skip
+  
+  ENV.append "LDFLAGS", "-headerpad_max_install_names"
```
Comment (mikemcquaid) [inline gcc48.rb]: Needs to be inside the `install` method.

Comment (DomT4) [inline gcc48.rb]: Welp. How didn't I notice that. Thanks for the spot!

**Comment by DomT4** [comment]
Still failing on ML, alas. Garg.

```
error: /usr/bin/install_name_tool: changing install names or rpaths can't be redone for: /usr/local/Cellar/gcc48/4.8.4/libexec/gcc/x86_64-apple-darwin12.6.0/4.8.4/lto1 (for architecture x86_64) because larger updated load commands do not fit (the program must be relinked, and you may need to use -headerpad or -headerpad_max_install_names)
```

**Comment by DomT4** [comment]
Merged a version of this in in https://github.com/Homebrew/homebrew-versions/commit/39bf54ee10abe192228438257c4b5fac3693a751 - Kept your commit credit intact. Thanks for your contribution to Homebrew, we appreciate it! :smiley_cat:.

**Comment by leonardoobaptistaa** [comment]
Thanks for keeping the credit :)  
I am glad to help

**Pull request closed without merging**